examples/libh2o/latency-optimization.c

```diff
@@ -202,8 +202,10 @@ static void on_handshake_complete(h2o_socket_t *sock, const char *err)
 static void on_connect(h2o_socket_t *sock, const char *err)
 {
     if (err != NULL) {
+        char buf[128];
         /* connection failed */
-        fprintf(stderr, "failed to connect to host:%s\n", err);
+        fprintf(stderr, "failed to connect to host:%s:%s\n", err,
+                h2o_strerror_r(errno, buf, sizeof(buf)));
```

Comment:
I think this change is also not appropriate due to the same reason (i.e., the failure might be due to something other than a syscall, e.g., certificate validation).

Reply:
Seems 'on_connect' only called after connection ready before ssl handshake, am i misunderstanding something?